### DISCUSSION TITLE
"Hide until" parameter

### DISCUSSION THREAD

**Discussion by t214c** [iSoron/uhabits#1732] [Feature Requests]
With many habits, the UI can be cluttered and confusing.

I have a specific time-of-day reminder for most habits. It would be awesome to be able to hide (filter out of the list) a habit until certain time; absolute or relative (unhide 2 hours before due time).

This way, when I wake up and check my phone, I don't need to see before-bedtime habits.

Thanks

**Comment by navinthomas** [comment]
Why not rearrange the habits manually so that habits like your before-bedtime habits are shown at the end rather than at the beginning?

A technique that I use is to set the sort to "By color". I then assign each habit a color that will make them appear higher up in the list if it's a morning habit, or lower in the list if it's an evening or night habit. For more info, check this discussion post -> https://github.com/iSoron/uhabits/discussions/1343

**Comment by t214c** [comment]
@navinthomas I already do. But I have many habits that are not time-specific (can be done in the morning or at night) and many that are. It is the latter that I want to hide for a cleaner look and easier tracking. But thanks for the advice.

**Reply by navinthomas** [reply]
So, you want the habits that are not time specific to be visible all the time, and the habits that are time-specific to be visible after the habits' due time, right? In that case, I can understand your use case much more clearer.

**Reply by t214c** [reply]
More or less. I'd personally show it an hour or so before due time.
